### ISSUE TITLE
[d3d8] Railroad Tycoon 3 - zoomed in terrain shuffle

### ISSUE THREAD

**Issue by WinterSnowfall** [doitsujin/dxvk#4114]
This is going to be a rather hard one to describe, so [I've captured a video](https://drive.proton.me/urls/VCR9YNK5F8#Ft8HtL438gFy). The TLDR of it is that when scrolling through some of the closest zoom levels on the game map, the terrain textures sort of shuffle around in a crazy dance.

I've captured a trace, but since the game does "wonderful" things with buffers, apitrace has mostly caught garbage. The issue can be seen in the trace as well though, among the madness.

### Software information
Railroad Tycoon 3

### System information
- GPU: RTX 4070
- Driver: 535.183.01
- Wine version: 9.12 Staging
- DXVK version: fe9c875

### Apitrace file(s)
[RT3.trace.tar.xz](https://drive.proton.me/urls/852R9WZMW0#nMgxRmw1VguQ)

### Log files
[RT3.log](https://github.com/AlpyneDreams/d8vk/files/12549377/RT3.log)

**Comment by WinterSnowfall** [comment]
Could be an ANV bug, but on Intel the game doesn't render geometry properly at all and ultimately fails with:

`/usr/i686-w64-mingw32/include/c++/13.1.0/array:202: constexpr std::array<_Tp, _Nm>::value_type& std::array<_Tp, _Nm>::operator[](size_type) [with _Tp = dxvk::DxsoSemantic; unsigned int _Nm = 32; reference = dxvk::DxsoSemantic&; size_type = unsigned int]: Assertion '__n < this->size()' failed.`

**Comment by WinterSnowfall** [comment]
New set of logs, collected with a debug build from @K0bin: [RT3.log](https://github.com/user-attachments/files/17062470/RT3.log)

**Comment by K0bin** [comment]
We just fixed the Intel crash at least.

**Comment by K0bin** [comment]
Apparently it crashes on ANV inside `vkCreateGraphicsPipeline` now.

There are a bunch of validation errors that need to get investigated.
[RT3 validation errors.txt](https://github.com/user-attachments/files/17065259/RT3.validation.errors.txt)

**Comment by doitsujin** [comment]
This should work with dxbc-spirv.

**Comment by WinterSnowfall** [comment]
Nope, still doing the terrain shuffle (see the large patch at the bottom).

<img width="1280" height="800" alt="Image" src="https://github.com/user-attachments/assets/7beddb5d-7d77-4911-95c6-0fcf1a32cf56" />

**Comment by doitsujin** [comment]
apparently I misunderstood the issue because the trace is borked and all the discussion was around an ANV crash.

Anyway, debugged this now and the game is requiring fixed-function vertex shaders to **produce identical math to their own custom programmable shaders**, which is completely insane.
